src/sinks/aws_sqs.rs

```diff
@@ -0,0 +1,412 @@
+            client,
+            queue_url: config.queue_url,
+        };
+
+        let sink = request
+            .batch_sink(
+                SqsRetryLogic,
+                sqs,
+                VecBuffer::new(batch.size),
+                batch.timeout,
+                cx.acker(),
+            )
+            .sink_map_err(|error| error!(message = "Fatal sqs sink error.", %error))
+            .with_flat_map(move |event| {
+                stream::iter(encode_event(event, &encoding, message_group_id.as_ref())).map(Ok)
+            });
+
+        Ok(sink)
+    }
+}
+
+impl Service<Vec<SendMessageEntry>> for SqsSink {
+    type Response = SendMessageResult;
+    type Error = RusotoError<SendMessageError>;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, mut entries: Vec<SendMessageEntry>) -> Self::Future {
+        assert_eq!(entries.len(), 1, "Sending batch is not supported.");
+
+        let entry = entries.remove(0);
+        let byte_size = entry.message_body.len();
+
+        let client = self.client.clone();
+        let request = SendMessageRequest {
+            message_body: entry.message_body,
+            message_group_id: entry.message_group_id,
+            queue_url: self.queue_url.clone(),
+            ..Default::default()
+        };
+
+        Box::pin(async move {
+            client
+                .send_message(request)
+                .inspect_ok(|result| {
+                    emit!(AwsSqsEventSent {
+                        byte_size,
+                        message_id: result.message_id.as_ref()
+                    })
+                })
+                .instrument(info_span!("request"))
+                .await
+        })
+    }
+}
+
+#[derive(Debug, Clone)]
+struct SendMessageEntry {
+    message_body: String,
+    message_group_id: Option<String>,
+}
+
+impl EncodedLength for SendMessageEntry {
+    fn encoded_length(&self) -> usize {
+        self.message_body.len()
+    }
+}
+
+impl Response for SendMessageResult {}
+
+#[derive(Debug, Clone)]
+struct SqsRetryLogic;
+
+impl RetryLogic for SqsRetryLogic {
+    type Error = RusotoError<SendMessageError>;
+    type Response = SendMessageResult;
+
+    fn is_retriable_error(&self, error: &Self::Error) -> bool {
+        match error {
+            RusotoError::HttpDispatch(_) => true,
+            RusotoError::Unknown(res) if res.status.is_server_error() => true,
+            _ => false,
+        }
+    }
+}
+
+fn encode_event(
+    mut event: Event,
+    encoding: &EncodingConfig<Encoding>,
+    message_group_id: Option<&Template>,
+) -> Option<SendMessageEntry> {
+    encoding.apply_rules(&mut event);
+
+    let message_group_id = message_group_id.and_then(|tpl| match tpl.render_string(&event) {
+        Ok(value) => Some(value),
+        Err(missing_keys) => {
+            error!(
```

Comment:
We should emit events for this instead, similar to the elasticsearch sink:

https://github.com/timberio/vector/blob/2d4fa207f460de5f1c6f08bad05bd0c3aa3a36d1/src/sinks/elasticsearch.rs#L183-L185